modules/GO_TOPGO.R

```diff
@@ -0,0 +1,277 @@
+### Load package and set path
+library("topGO")
+library("org.Mm.eg.db")
+library("ggplot2")
+
+
+
+
+#### Load data
+
+## from file
+#data <- data.table::fread("RESULTS_DEG_HNSC.txt")
+#data$GeneID <- substr(data$GeneID, 1, 18)
+
+
+## from web-source
+data <- data.table::fread("https://raw.githubusercontent.com/ycl6/GO-Enrichment-Analysis-Demo/master/DESeq2_DEG.txt")

```

Comment:
The path should become a parameter of your script. The easiest way to achieve this is to use `commandArgs` as described [here](https://www.r-bloggers.com/2015/09/passing-arguments-to-an-r-script-from-command-lines/). 

A bit more advanced options (mostly useful if you have many arguments) are argparse as used by Sandro in [the deseq2 module](https://github.com/icbi-lab/nf-deseq2/blob/c56404f184146ae144b9babc6edc1cffe182c8af/bin/DESeq2_script.R#L15-L33) and [docopt](https://github.com/docopt/docopt.R) as used by myself [here](https://github.com/icbi-lab/nf-deseq2/blob/fdc7b21e4ccc1723618a0a1250059c2e642b7374/bin/clusterprofiler_ora.R#L3-L20).

Reply:
#Get input arguments
arg = commandArgs(trailingOnly=TRUE)

#Test if there is a inputfile argument
if (length(args)!=1) {
  stop("You have to enter the full path to your source file as parameter!", call.=FALSE)
}

-> it always make an Error

Reply:
How do you call the script when the error occurs, and what's the exact error message?

Reply:
i deleted the test....